### ISSUE TITLE
Showing Error while make

### ISSUE THREAD

**Issue by krishnakanth12** [ddlawrence/BBB_UART_IRQ#2]
while doing make showing this error : make: *** No rule to make target `slib\startup.s', needed by `olib\startup.o'.  Stop.

So if possible can you please help me 
Thank you

**Comment by ddlawrence** [comment]
Hi.  The Makefile rule   olib\startup.o : slib\startup.s
expects to see a file startup.o in directory olib.
So, copy any file there and name it startup.o
Then modify startup.s so that it's timestamp is newer.  
The rule should execute.....

**Comment by krishnakanth12** [comment]
Thanks for Replying Sir, And i can not find olib in your BBB_UART_IRQ file or can you please tell me where directory olib is located. 
Thank you

**Comment by ddlawrence** [comment]
Yes, you are right.  the github repo does not have slib and olib directories.  
i just uploaded them.  so, redownload the repo from here or www.baremetal.tech

thanks for pointing this out...........dd

**Comment by krishnakanth12** [comment]
Thanks for replying back, And i am facing new error while make like root@beaglebone:/home/debian/git/Compressed# make
arm-none-eabi-as startup.s -o startup.o
make: arm-none-eabi-as: Command not found
make: *** [startup.o] Error 127
 
can you suggest me to avoid and successfully run the make file 
Thank you.

**Comment by krishnakanth12** [comment]
In MAKEfile what is the path for CROSSCOMPILE for Linux please help me Sir.

**Comment by ddlawrence** [comment]
hi.  so u r developing on a linux machine
the first line of the makefile is the compiler name 
CROSSCOMPILE = arm-linux-gnueabihf-

go to www.baremetal.tech and get the noob_guide
it is a little out of date, but covers alot of what u need 2 know

later............dd
ps u can email me directly   d7 at hushmail dot com

**Comment by ddlawrence** [comment]
arm-none-eabi-as is for cross compiling in windows.  
use the other assembler listed in the makefile.............dd

**Comment by krishnakanth12** [comment]
yeah i did Sir but getting error like below  
root@beaglebone:/home/debian/git/Compressed# make
make: Warning: File `memmap.lds' has modification time 1.2e+03 s in the future
arm-linux-gnueabihf- as startup.s -o startup.o
make: arm-linux-gnueabihf-: Command not found
make: *** [startup.o] Error 127

and my device having installed compiler in /usr/sbin/  directory like 
 
arm-linux-gnueabihf-cpp       
arm-linux-gnueabihf-cpp-4.6    
arm-linux-gnueabihf-g++         
arm-linux-gnueabihf-g++-4.6    
arm-linux-gnueabihf-gcc        
arm-linux-gnueabihf-gcc-4.6     
arm-linux-gnueabihf-run 

i think i need to give path for compiler. please help me Sir Thank you

**Comment by ddlawrence** [comment]
hi.  see below

On 2/15/2018 at 8:55 AM, "Krishnakanth M" <notifications@github.com> wrote:
>
>yeah i did Sir but getting error like below  
>root@beaglebone:/home/debian/git/Compressed# make
>make: Warning: File `memmap.lds' has modification time 1.2e+03 s 
>in the future
>arm-linux-gnueabihf- as startup.s -o startup.o

in the above line there is a space in   arm-linux-gnueabihf-as before the "as"

the makefile and memmap looks like they've been corrupted or modified.  
redownload the zipfile from www.baremetal.tech and unzip it correctly.  

i apologize for these initial difficulties.  gnu and linux are not so well defined and documented.    
but they are free and will work once you get around these little problems.  
keep trying..................dd

ps if you need the gnu assembler for linux run:
apt-get install arm-linux-gnueabihf-as

**Comment by ddlawrence** [comment]
hi kris.  did u get the demo running yet?
